extensions/src/main/java/org/apache/falcon/extensions/store/ExtensionStore.java

```diff
@@ -269,6 +269,18 @@ public String registerExtension(final String extensionName, final String path, f
         throws URISyntaxException, FalconException {
         Configuration conf = new Configuration();
         URI uri = new URI(path);
+        if (uri.getScheme() == null){
```

Comment:
Actually, yes. It is always better to be as specific as possible. My concern here was more on redundant code. I guess that can be solved by introducing a method like the one below:
```
    private void assertURI(String part, String value) throws ValidationException {
        if (value == null) {
            String msg = "Invalid Path supplied. " + part + " is missing. "
                    + " Path must contain scheme, authority and path.";
            LOG.error(msg);
            throw new ValidationException(msg);
        }
    }
```